platform/android/export/export.cpp

```diff
@@ -2418,16 +2474,40 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 				store_in_gradle_project(img_path, data, Z_NO_COMPRESSION);
 			}
 		}
+	}
 
-        //TODO: replicate the functionality of _fix_manifest
+	Error gradle_build(const Ref<EditorExportPreset> &p_preset, bool p_debug, const String &p_path, String &src_apk, int p_flags = 0) {
 
-        //        String file = "res://android/build/AndroidManifest.xml";
-        //
-        //        Vector<uint8_t> android_manifest_file_data = FileAccess::get_file_as_array(file);
-        //        _fix_manifest(p_preset, android_manifest_file_data,  p_flags & (DEBUG_FLAG_DUMB_CLIENT | DEBUG_FLAG_REMOTE_DEBUG));
+		{ //test that installed build version is alright
+			FileAccessRef f = FileAccess::open("res://android/.build_version", FileAccess::READ);
+			if (!f) {
+				EditorNode::get_singleton()->show_warning(TTR("Trying to build from a custom built template, but no version info for it exists. Please reinstall from the 'Project' menu."));
+				return ERR_UNCONFIGURED;
+			}
+			String version = f->get_line().strip_edges();
+			if (version != VERSION_FULL_CONFIG) {
+				EditorNode::get_singleton()->show_warning(vformat(TTR("Android build version mismatch:\n   Template installed: %s\n   Godot Version: %s\nPlease reinstall Android build template from 'Project' menu."), version, VERSION_FULL_CONFIG));
+				return ERR_UNCONFIGURED;
+			}
+		}
+		//build project if custom build is enabled
```

Comment:
Is this comment correct? I thought this code was already within the "custom build" section